packages/ods/src/components/pagination/src/controller/ods-pagination.ts

```diff
@@ -0,0 +1,80 @@
+import type { OdsPaginationPageList } from '../interfaces/pagination-page-list';
+
+function computeActualTotalPages(itemPerPage: number, totalItems: number | undefined, totalPages: number): number {
+  if (!totalItems) {
+    return totalPages;
+  }
+
+  return Math.ceil(totalItems / (itemPerPage || 1));
+}
+
+function createPageList(totalPages: number, pageSelected: number): OdsPaginationPageList {
+  const MAX_VISIBLE_ITEMS = 7;
+  const ELLIPSIS_THRESHOLD = 4;
+  const MINIMUM_PAGE = 1;
+  const DEFAULT_PAGE_OFFSET = 2;
```

Comment:
declare the constants outside of any functions, export them at the end of the file and use them in the component